src/Glpi/Agent/Communication/AbstractRequest.php

```diff
@@ -207,6 +210,22 @@ public function handleHeaders()
      */
     public function handleRequest($data): bool
     {
+        $auth_required = \Config::getConfigurationValue('inventory', 'auth_required');
+        if ($auth_required === 'client_credentials') {
+            $request = new Request('POST', $_SERVER['REQUEST_URI'], $this->headers->getHeaders());
+            try {
+                $client = Server::validateAccessToken($request);
+                if (!\User::isNewID($client['users_id']) || !in_array('inventory', $client['scopes'], true)) {
```

Comment:
Does the `!\User::isNewID($client['users_id'])` part of this check means that we are not in the case of a `client_credentials` token? What is the value returned in this case in `$client['users_id']`?

By the way, the key to check is probably `user_id`, according to https://github.com/glpi-project/glpi/blob/b820eba628e37db48f7d92f66da63c120b2e1ef0/src/Glpi/OAuth/Server.php#L147

```suggestion
                if (!\User::isNewID($client['user_id']) || !in_array('inventory', $client['scopes'], true)) {
```

Reply:
My comment is still valid.